### ISSUE TITLE
leap-forward-x and selection=exclusive

### ISSUE THREAD

**Issue by nkgm** [ggandor/leap.nvim#52]
Right now `leap-forward-x` lands 1 char short in visual mode when `selection=exclusive`. I'm assuming this isn't intended behavior as `leap-forward` does the right thing.

And thanks for this plugin :)

**Comment by ggandor** [comment]
I don't understand the issue, they work consistently. Targeting `ar`:
`leap-forward` + `sel=exclusive` -> `foob|ar`
`leap-forward` + `sel=inclusive` -> `fooba|r`
`leap-forward-x` + `sel=exclusive` -> `fooba|r`
`leap-forward-x` + `sel=inclusive` -> `foobar|`

> And thanks for this plugin :)

Thanks :)

**Comment by nkgm** [comment]
I would like to contrast this behavior to vim's own `t`, `f`:

`vta` + `sel=exclusive` -> `foob|ar`
`vta` + `sel=inclusive` -> `foo|b|ar`
Effective selection is `foob` in both cases.

`vfa` + `sel=exclusive` -> `fooba|r`
`vfa` + `sel=inclusive` -> `foob|a|r`
Effective selection is `fooba` in both cases.

I would expect Leap to follow the same principle, so that `leap-forward + ar` is always `foob|ar` and `leap-forward-x + ar` is always `foobar|`.

**Comment by ggandor** [comment]
> vta + sel=exclusive -> foob|ar
> vta + sel=inclusive -> foo|b|ar
> Effective selection is foob in both cases.

No. `vta + sel=exclusive` is `foo|bar` (it would yank `foo`), `vfa + sel=exclusive` is foob|ar (yanks `foob`). What would be the point of the `selection` flag otherwise?

**Comment by nkgm** [comment]
> vta + sel=exclusive is foo|bar (it would yank foo)

![vim vta](https://user-images.githubusercontent.com/5631881/192239707-56382563-3f35-4627-a50c-c019c07a3a5c.gif)
`vta` always yanks `foob` regardless of `sel`.

> vfa + sel=exclusive is foob|ar (yanks foob)

![vim vfa](https://user-images.githubusercontent.com/5631881/192241437-7aefaccd-d852-49cc-bcc6-9d694425ed1a.gif)
`vfa` always yanks `fooba` regardless of `sel`.

**Comment by ggandor** [comment]
I stand corrected. I tried this on vanilla Vim, but on Nvim `t` indeed behaves as you described.

I cannot see this documented anywhere though. I've read though `:h vim-diff.txt` and `:h visual.txt`, and also tried to search for relevant Nvim issues.

**Comment by nkgm** [comment]
> I tried this on vanilla Vim

Are you sure? I tried this on vanilla Vim and it produced the same yanks (the caret moves an extra char to the right under `sel=exclusive` but it's the same yank).

Either way this issue is more of an exercise in Vim as I'm well aware I can do my custom motions in Leap thanks to its excellent API, however every now and then I find myself indulging in conversation about such aspects of Vim as some things still fail to make sense to me. For instance `dw` and `vwd` will give different results under `sel=inclusive`. I'm sure there's "reasons", but from a usability perspective this is insane as you have to keep the various edge cases in mind depending on where your are in a file and what mode you're in, what direction you're going etc (end-of-line is another favorite). It would make way more sense to always use `sel=exclusive` with `ve=onemore` with a custom `e`-like motion mapped to a "smart" `el` (to always land 1 char past the end of a word so that a subsequent `vb` highlights the entire word), but I digress!
